### PR TITLE
frontend: Fix bug with using same x-axis for all metrics

### PR DESCRIPTION
**Pull request by punchagan**
963997b3f0c87147774c04899ac5b0fdd30574a9 added code to ensure that all the
metrics used the same X-axis, even if some metrics didn't have data for
specific commits. The commit incorrectly used the same metadata for all the
commits, and metadata contained other important information like units, trend,
description, etc. This commit fixes the change to use correct metadata for all
the commits for each metric. The metadata from the last commit is used as a
template and the commit and runAt values are changed.

### CODE REVIEW

File: frontend/src/AppHelpers.res [hunk 1]
```diff
@@ -31,34 +31,48 @@ let sortedMetadata = dataByTestName => {
...
+      // We use the last metadata item and use that as a template to create
+      // entries for missing commits. commit, runAt, lines and run_job_id are
+      // overwritten.  Other metadata like units, trend, description are the
+      // same as metadata in the last commit.
+      let templateMetadata = BeltHelpers.Array.lastExn(metricMetadata)
+      let filledMetadata =
+        allCommits->Belt.Array.map(((commit, runAt, run_job_id)) =>
+          Js.Obj.assign(
+            templateMetadata,
+            {"commit": commit, "runAt": runAt, "run_job_id": run_job_id, "lines": []},
```
Comment (art-w) [inline frontend/src/AppHelpers.res:61]: I'm not sure why we have a javascript object to represent all these stuff (I think `BenchmarkData.res` could have defined a standard record)... But now it's a bit everywhere and it's not really convenient.

For example, I don't see a nice alternative here to `Js.Obj.assign` and the issue is that this function breaks the typechecker (it returns an unknown object that could contain anything -- ie it's as bad as `Obj.magic`)

**Review by art-w — APPROVED** [review]
Thanks for the fix! My comment about `Js.Obj.assign` is not a blocker, more a note for later (unless you really really want to work on an alternative right now), feel free to merge!

File: frontend/src/AppHelpers.res (hunk 1 above)
Comment (punchagan) [reply frontend/src/AppHelpers.res:61]: I agree that it is annoying that the metric defined in `BenchmarkData.res` is not a standard record. I've felt that having a record would've been a lot more convenient atleast a couple of times. We should refactor that.

**Comment by punchagan** [comment]
> My comment about `Js.Obj.assign` is not a blocker, more a note for later (unless you really really want to work on an alternative right now), feel free to merge!

I'll create a new issue for tracking this, and merge the current PR since it's a very visible bug for users if we end-up deploying what's already been merged.

**Pull request merged**